mkosi/__init__.py

```diff
@@ -388,6 +388,64 @@ def configure_os_release(context: Context) -> None:
         if ArtifactOutput.os_release in context.config.split_artifacts:
             shutil.copy(osrelease, context.staging / context.config.output_split_os_release)
 
+        if ArtifactOutput.metainfo in context.config.split_artifacts:
+            # https://www.freedesktop.org/software/appstream/docs/sect-Metadata-OS.html
+            metainfo = '<?xml version="1.0" encoding="UTF-8"?>\n'
+            metainfo += '<component type="operating-system">\n'
+            id = name = home_url = pretty_name = None
+            for line in osrelease.read_text().splitlines():
+                if line.startswith("ID="):
+                    id = line.split("=")[1].strip('"')
+                elif line.startswith("NAME="):
+                    name = line.split("=")[1].strip('"')
+                elif line.startswith("HOME_URL="):
+                    home_url = line.split("=")[1].strip('"')
+                elif line.startswith("PRETTY_NAME="):
+                    pretty_name = line.split("=")[1].strip('"')
+
+            # The <id> format is reverse DNS style with home_url + id
+            if home_url:
+                from urllib.parse import urlparse
+
+                url = urlparse(home_url)
+                if url.netloc:
+                    netloc = url.netloc.split(".")
+                    netloc.reverse()
+                    netloc.remove("www")
```

Comment:
This removes any www, not only one at the start

Reply:
www is not a valid tld or domain

Reply:
But it is a valid subdomain (e.g. `distro.www.company.org`). Not sure who would do that stuff but there is no reason to not just do the correct thing.

```suggestion
                    netloc = url.netloc.removeprefix("www.").split(".")
```